database/engine/rrdengine.c

```diff
@@ -421,11 +418,67 @@ void read_extent_cb(uv_fs_t* req)
     }
     if (xt_io_descr->completion)
         completion_mark_complete(xt_io_descr->completion);
+}
+
+static void read_extent_cb(uv_fs_t *req)
+{
+    struct rrdengine_worker_config *wc = req->loop->data;
+    struct extent_io_descriptor *xt_io_descr;
+
+    xt_io_descr = req->data;
+    do_extent_processing(wc, xt_io_descr, req->result < 0);
     uv_fs_req_cleanup(req);
     posix_memfree(xt_io_descr->buf);
     freez(xt_io_descr);
 }
 
+static void read_mmap_extent_cb(uv_work_t *req, int status __maybe_unused)
+{
+    struct rrdengine_worker_config *wc = req->loop->data;
+    struct rrdengine_instance *ctx = wc->ctx;
+    struct extent_io_descriptor *xt_io_descr;
+    xt_io_descr = req->data;
+
+    if (likely(xt_io_descr->map_base)) {
+        do_extent_processing(wc, xt_io_descr, false);
+        munmap(xt_io_descr->map_base, xt_io_descr->map_length);
+        freez(xt_io_descr);
+        return;
+    }
+
```

Comment:
Is it worth adding a
```
#ifdef NETDATA_INTERNAL_CHECKS 
debug(D_RRDENGINE, .....
```
to check the `errno` in case mmap fails?

Reply:
This will not be very useful wrapped in the debug flag since there is also a fallback. In practice mmap failing would indicate that something is about to fail anyway. Notice the fallback will also try to allocate memory.

Reply:
OK, I was thinking mostly for us to see the reason why `mmap` would fail.